src/app/game-state/impossibleTask.service.ts

```diff
@@ -170,7 +170,16 @@ export class ImpossibleTaskService {
   }
 
   reset(){
+    let completeTaskIndex = -1
+    for (let taskIndex = 0; taskIndex < this.tasks.length + 1; taskIndex++){
+      if (this.taskProgress[taskIndex].complete){
+        completeTaskIndex = taskIndex; // Sanity check for the latest complete task
+      }
+    }
     for (let taskIndex = 0; taskIndex < this.tasks.length; taskIndex++){
+      if (taskIndex < completeTaskIndex){
```

Comment:
Thought: If we loop through backwards we could do this in one loop.

Reply:
Thought in reverse, I should have. I'll update.

Reply:
Oop, and testing again with loading an old save caused an issue, time to find and fix.